### ISSUE TITLE
如何在你训练的基础上增加训练集？

### ISSUE THREAD

**Issue by Steverdeng** [chineseocr/chineseocr#43]
请问一下，我测试发现ⅡⅢⅣⅤⅥⅠ这些字很难识别出来，请问我如何在你的基础上增加训练这些类型的字？

**Comment by wenlihaoyu** [comment]
可以进行fine-tune，或者进行迁移学习。或者根据应用场景修正，比如你的应该场景只用到“1234567890”,那么在crnn模型输出的概率矩阵环节，应该先过滤后，再进行后续处理。

**Comment by Steverdeng** [comment]
这几个字ⅡⅢⅣⅤⅥ 在 代码crnn/keys.py是找不到的。就是不知道如何添加训练

**Comment by Steverdeng** [comment]
你训练出来模型汉字是比较好的，就是英文以及一些字符不太好

**Comment by wenlihaoyu** [comment]
修改模型的输出层，重新训练即可。

**Comment by ZJU-PLP** [comment]
@wenlihaoyu 您好,想问一下fine-tune是将你生成的模型作为预训练模型,然后再训练我们自己的数据集吗?另外如果我想制作自己的label,要怎样进行啊?还有你说的crnn模型输出的概率矩阵环节，应该先过滤后，再进行后续处理,不是很理解这句话啊

**Comment by Aurora11111** [comment]
@wenlihaoyu the same ,how to fine-tune

**Comment by xxllp** [comment]
这个能识别证件照 啥的嘛

**Comment by Steverdeng** [comment]
是不是根据你的代码，我只需要训练我自己那部分字符就可以了？

**Comment by wenlihaoyu** [comment]
@Steverdeng 对

**Comment by beimingmaster** [comment]
这个模型手写体识别支持吗？能否增加手写体数据训练？

**Comment by beimingmaster** [comment]
> @beimingmaster 部分手写体可以识别，我测试过了，不知道你的应用场景是怎样的，可以跑作者的模型试一下

谢谢，手写体我看网上有中科院的数据，我打算加点训练数据试试。
看了下前面的issue，好像没有一个明确说明如何增加额外训练数据的描述文档。

**Comment by xiaosi2017** [comment]
请问：我把作者训练好的模型作为预训练模型，然后在自己的两百多万张数据集训练这样会导致过拟合吗，加了early stopping；
另外请教一个基础的问题，如果我先训练了5个epoch，得到一个模型，在用这个模型作为预训练模型再训练5个epoch，和直接训练10个epoch结果有差别很大吗？

**Comment by mocorr** [comment]
@qq542288564 如果你是想增加一些原keys.py没有的字符，追加在最后面就好了。把nclass的数量改成你的字符集总数。你可以从头自己训练，或者在作者的权重文件的基础上，只对最后一层训练。我猜想你对神经网络并不了解，建议先自学一下基本的分类网络和迁移学习。

**Comment by Aurora11111** [comment]
@OKhyc reference:https://github.com/Aurora11111/crnn-train-pytorch

**Comment by Aurora11111** [comment]
@OKhyc 是的啊

**Comment by bournes** [comment]
如何配合这个使用呢！

**Comment by bournes** [comment]
如何配合这个使用呢！ https://github.com/chineseocr/ocr-label

**Comment by zlszhonglongshen** [comment]
@Aurora11111  你上面提供crnn非常好，我想问下，该如何训练文字检测呢？

**Comment by Aurora11111** [comment]
reference:https://github.com/Aurora11111/text-detection-train-yolov3



在 2019-11-01 16:16:16，"zlszhonglongshen" <notifications@github.com> 写道：


@Aurora11111 你上面提供crnn非常好，我想问下，该如何训练文字检测呢？

**Comment by DominicTerry** [comment]
@wenlihaoyu 我是想在您原有的pytorch ocr识别模型ocr-lstm.pth基础上，再增加自己的训练集，自己准备好了行图片与行文本，训练代码参考只有https://github.com/Aurora11111/crnn-train-pytorch吗？里面有不少坑，比如不用warp_ctc_pytorch，直接用新版本的torch.nn.CTCLoss 函数可不可以。有没有可参考的最新的训练代码，直接载入原来的模型，然后加上自己的数据集行图片与行文本，生成新的模型文件。

**Comment by Zhang-O** [comment]
> > @beimingmaster 部分手写体可以识别，我测试过了，不知道你的应用场景是怎样的，可以跑作者的模型试一下
> 
> 谢谢，手写体我看网上有中科院的数据，我打算加点训练数据试试。
> 看了下前面的issue，好像没有一个明确说明如何增加额外训练数据的描述文档。

请问使用中科院的手写数据后，对手写支持的怎么样呢
